### PR TITLE
test(ref): test for resolving ref for security scheme in Asyncapi3

### PR DESCRIPTION
**Pull request by cka121**
This PR adds unit tests for resolving reference for security scheme object in AsyncAPI 3 [#OSS-370](https://smartbear.atlassian.net/browse/OSS-370)

### My PR contains...

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [X] are not breaking changes.

### Documentation
- [ ] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [X] All new and existing tests passed.

### CODE REVIEW

File: packages/apidom-reference/test/dereference/strategies/asyncapi-3/security-scheme-object/fixtures/operation-object/root.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  "asyncapi": "3.0.0",
+  "components": {
+    "operations": {
+      "operation1": {
+        "security": {
+          "type": "string"
+        }
+      },
+      "operation2": {
+        "$ref": "#/components/operations/operation1"
```
Comment (glowcloud) [inline packages/apidom-reference/test/dereference/strategies/asyncapi-3/security-scheme-object/fixtures/operation-object/root.json]:
```suggestion
        "security": [{
          "$ref": "#/components/operations/operation1/security/0"
        }]
```


And I think we will need to fix the Visitor, similar to ServerSecurityVisitor: https://github.com/swagger-api/apidom/blob/main/packages/apidom-ns-asyncapi-3/src/refractor/visitors/async-api-3/operation/SecurityVisitor.ts#L33

File: packages/apidom-reference/test/dereference/strategies/asyncapi-3/security-scheme-object/fixtures/operation-object/root.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "asyncapi": "3.0.0",
+  "components": {
+    "operations": {
+      "operation1": {
+        "security": {
```
Comment (glowcloud) [inline packages/apidom-reference/test/dereference/strategies/asyncapi-3/security-scheme-object/fixtures/operation-object/root.json]: Should be an array of Security Schemes
```suggestion
        "security": [{
```

File: packages/apidom-reference/test/dereference/strategies/asyncapi-3/security-scheme-object/fixtures/operation-trait-object/root.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "asyncapi": "3.0.0",
+  "components": {
+    "operationTraits": {
+      "operationTrait1": {
+        "bindings": {
```
Comment (glowcloud) [inline packages/apidom-reference/test/dereference/strategies/asyncapi-3/security-scheme-object/fixtures/operation-trait-object/root.json]:
```suggestion
        "security": [{
           "type": "apiKey"
        }]
```

File: packages/apidom-reference/test/dereference/strategies/asyncapi-3/security-scheme-object/fixtures/operation-trait-object/root.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+          "kafka": {}
+        }
+      },
+      "operationTrait2": {
+        "$ref": "#/components/operationTraits/operationTrait1"
```
Comment (glowcloud) [inline packages/apidom-reference/test/dereference/strategies/asyncapi-3/security-scheme-object/fixtures/operation-trait-object/root.json]:
```suggestion
        "security": [{
          "$ref": "#/components/operationTraits/operationTrait1/security/0"
        }]
```

File: packages/apidom-reference/test/dereference/strategies/asyncapi-3/security-scheme-object/fixtures/server-object/root.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+  "asyncapi": "3.0.0",
+  "servers": {
+    "server1": {
+			"security":  [{ "$ref": "#/SecuritySchemes" }]
```
Comment (glowcloud) [inline packages/apidom-reference/test/dereference/strategies/asyncapi-3/security-scheme-object/fixtures/server-object/root.json]: Reference is correct, just a potential change, if others think we should reference a field defined by AsyncAPI, and not an additional one:
```suggestion
    "server1": {
			"security":  [{ "$ref": "#/servers/server2/security/0" }]
     },
    "server2": {
			"security":  [{ "type": "apiKey" }]
     }
```
